code/drasil-example/nopcm/lib/Drasil/NoPCM/GenDefs.hs

```diff
@@ -14,5 +14,5 @@ genDefs = [rocTempSimp, htFluxWaterFromCoil]
 
 rocTempSimp :: GenDefn
 rocTempSimp = gdNoRefs (othModel' rocTempSimpRC) (Nothing :: Maybe UnitDefn)
-  (Just $ rocTempSimpDeriv EmptyS [assumpCWTAT, assumpDWCoW, assumpSHECoW])
+  (Just $ rocTempSimpDeriv rocTempSimp EmptyS [assumpCWTAT, assumpDWCoW, assumpSHECoW])
```

Comment:
This feels like an orthogonal (but good) change, where additional information is being attached to the derivation. Is that the case?

Reply:
It's an orthogonal issue, yes, but it's at least making the problem known (I don't think it was known before?). I think we're still losing out on information that we might want to retain later via refinement (here, showing that NoPCM's variant of rocTempSimp is that of SWHS with some alterations).

Since NoPCM's rocTempSimp originates from SWHS' but they use different ModelKinds, when I changed how `othModel'` alters UIDs, this caused an unknown UID "Reference" error for `rocTempSimpRC` (which is the existing UID of the same model from SWHS, but in NoPCM, it's UID is `rocTempSimpRCMK`) because SWHS' rocTempSimp's derivation had a reference to itself! However, since _that_ version of `rocTempSimp` is not the same one that NoPCM has, it caused a reference bug (since it's not registered in NoPCM's ChunkDB).

As such, I made his temporary measure to allow us to have self-referencing in chunk contents without causing friction. It might be better to create some sort of "SelfReference" data type, but that would need design. This should be better transcribed when we have a full refinement/specialization system.

Reply:
Can you also start a new issue for this? I feel like you're telling me about 1/2 of the information I really need to know to understand the issue. And definitely not enough information to understand if this solution is a fine hack or a horrible hack that shouldn't be let through.

Reply:
Sounds good. I might not be explaining it well enough without being able to refer to specific code snippets. I'll try to write a code-snippet-oriented explanation in a new ticket.